enhancements/network/ovn-kubernetes-localnet-api.md

```diff
@@ -0,0 +1,557 @@
+labels:
+  k8s.ovn.org/user-defined-network: ""
+ownerReferences:
+- apiVersion: k8s.ovn.org/v1
+  blockOwnerDeletion: true
+  controller: true
+  kind: ClusterUserDefinedNetwork
+  name: test-net
+  uid: 293098c2-0b7e-4216-a3c6-7f8362c7aa61
+spec:
+    config: > '{
+        "cniVersion": "1.0.0",
+        "type": "ovn-k8s-cni-overlay"
+        "netAttachDefName": "blue/test-net",
+        "role": "secondary",
+        "topology": "localnet",
+        "name": "cluster.udn.test-net",
+        "physicalNetworkName: "tenantblue",
+        "mtu": 1500,
+        "subnets": "192.168.100.0/24,2001:dbb::/64"
+    }'
+```
+Example 2 (custom MTU, VLAN and sticky IPs):
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: test-net
+spec:
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: In
+      values: ["red", "blue"]
+  network:
+    topology: Localnet
+    localnet:
+      role: Secondary
+      physicalNetworkName: tenantblue
+      subnets: ["192.168.0.0/16", "2001:dbb::/64"]
+      excludeSubnets: ["192.168.50.0/24"]
+      mtu: 9000
+      vlan: 200
+      ipam:
+        lifecycle: Persistent
+```
+The above spec will generate the following NAD, in namespace `red`:
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: test-net
+  namespace: red
+finalizers:
+ - k8s.ovn.org/user-defined-network-protection
+labels:
+  k8s.ovn.org/user-defined-network: ""
+ownerReferences:
+- apiVersion: k8s.ovn.org/v1
+  blockOwnerDeletion: true
+  controller: true
+  kind: ClusterUserDefinedNetwork
+  name: test-net
+spec:
+    config: > '{
+        "cniVersion": "1.0.0",
+        "type": "ovn-k8s-cni-overlay"
+        "netAttachDefName": "blue/test-net",
+        "role": "secondary",
+        "topology": "localnet",
+        "name": "cluster.udn.test-net",
+        "physicalNetworkName: "tenantblue",
+        "subnets": "192.168.0.0/16,2001:dbb::/64",
+        "allowPersistentIPs": true,
+        "excludeSubents: "10.100.50.0/24",
+        "mtu": 9000,
+        "vlanID": 200
+    }'
+```
+
+#### Go types
+The CUDN `spec.network.topology` field should be extended to accept `Localnet` string.
+And the `spec.network` struct `NetworkSpec` should be to have additional field for the localnet topology  configuration:
+```go
+const NetworkTopologyLocalnet NetworkTopology = "Localnet"
+
+...
+
+// NetworkSpec defines the desired state of UserDefinedNetworkSpec.
+// +union
+type NetworkSpec struct {
+    // Topology describes network configuration.
+    //
+    // Allowed values are "Layer3", "Layer2" and "Localnet".
+    // Layer3 topology creates a layer 2 segment per node, each with a different subnet. Layer 3 routing is used to interconnect node subnets.
+    // Layer2 topology creates one logical switch shared by all nodes.
+    // Localnet topology attach to the overlay local network. Enables egress to the provider's physical network. 
```

Comment:
I kind of always seen this the other way around; Localnet topology (which is an OVN term) attaches to the physical network .

Reply:
I changed the text saying localnet attach to physical network